hibernate-entitymanager/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -307,8 +316,12 @@ public boolean doExtendedEnhancement(CtClass classDescriptor) {
 	 */
 	private BootstrapServiceRegistry buildBootstrapServiceRegistry(
 			Map integrationSettings,
-			ClassLoader providedClassLoader) {
+			ClassLoader providedClassLoader, 
+			ClassLoaderService providedClassLoaderService) {
 		final BootstrapServiceRegistryBuilder bsrBuilder = new BootstrapServiceRegistryBuilder();
+		if (providedClassLoaderService != null) {
+			bsrBuilder.applyClassLoaderService(providedClassLoaderService);
```

Comment:
@sebersole, correct me if I'm wrong, but if this is applied, do we still need to do `bsrBuilder.applyClassLoader( providedClassLoader );` further down?  Since `providedClassLoader` is an `OsgiClassLoader`, isn't that already provided by the `ClassLoaderService`, and therefore `providedClassLoader` could be removed from the method sig?  Or are both needed and I'm missing a distinction?